enhancements/network/smart-nic-ovn-offload.md

```diff
@@ -0,0 +1,454 @@
+dataplane performance by offloading processing of OVS OpenFlow flows to
+hardware NIC. By integrating NVIDIA BlueField2 SmartNIC support in OpenShift,
+additional capabilities are added to accelerate and secure the OVN control
+plane by offloading OVS and OVN to the SmartNIC ARM processors.
+
+## Terminology
+
+- SmartNIC: A combined Network Interface Card (NIC) and CPU hardware which
+offloads processing tasks that the system CPU would normally handle
+- SmartNIC node(s): Baremetal servers where SmartNICs are plugged to
+- SmartNIC ARM cores: The embedded aarch64 system running on SmartNIC CPU(s)
+- Host-Network Pods: Pods that use networking on the host. OpenShift CNIs are
+not involved in configuring the networking for these pods. This allows the pods
+to access the host loopback device, services listening on localhost, and could
+be used to snoop on network activity of other pods on the same node.
+- SR-IOV: Single Root I/O Virtualization, a NIC technology that allows
+isolation of PCIe resources
+- SR-IOV Pods: SR-IOV Pods are pods bound to a VF which is used to send and
+receive packets on.
+- PF: Physical Function, the physical ethernet controller that supports SR-IOV
+- VF: Virtual Function, the virtual PCIe device created from PF. A given PF
+only supports a limited number of VFs (hardware dependant) and is in the order
+to 128 or 256 VFs per PF with NVIDIA BlueField2.
+- VF representor: The port representor of the VF. VF and VF representor are
+created in pairs where VF appears in SmartNIC node and VF representor appears
+in SmartNIC ARM cores
+
+## Motivation
+
+Encryption, Encapsulation, and Virtual switching (IPsec, Geneve, VXLAN, OVS,
+Connection Tracking) to and from containers or VMs are basic building blocks
+of the Hybrid cloud and particularly of OCP. These functions are very compute
+intensive and can consume up to 40 cores from the server’s motherboard CPU
+for terminating 25Gigs of traffic. Terminating 100Gigs without hardware offload
+is not feasible at all. Most of our customers are demanding all of the above
+without much use of the CPU on the motherboard of the server.
+
+### Goals
+
+- Only support NVIDIA BlueField2 because other implementations of SmartNIC are
+unknown
+- OVN-Kubernetes control plane including OVS and OVN controller are offloaded
+to SmartNIC, other CNIs are not currently planned
+- SR-IOV pods run on SmartNIC node using VF as pod default route interfaces
+- Manage SmartNIC node offload configurations in OpenShift Operators
+
+### Non-Goals
+
+- Provisioning and managing the OVN-Kubernetes components on SmartNIC ARM cores
+- Support multiple SmartNICs on one OpenShift baremetal worker node
+- Support SmartNICs on OpenShift baremetal master nodes
+
+## Proposal
+
+- User installs SR-IOV Operator (OLM managed)
+- User creates one or more custom MachineConfigPools whose nodes are installed
+with SmartNIC hardware
+- User specifies NodeSelector labels in SR-IOV Network Operator(SNO) CRD to
```

Comment:
I dont think we can use SNO in this context. It's already being used for Single Node OpenShift :)

Reply:
Updated to use SR-IOV Network Operator (removed SNO).